features/admin.server-configurations.v1/forms/username-recovery-form.tsx

```diff
@@ -100,34 +106,102 @@ export const UsernameRecoveryConfigurationForm: FunctionComponent<UsernameRecove
      * @param values - Form values.
      * @returns Sanitized form values.
      */
-    const getUpdatedConfigurations = (value: boolean) => {
-        const data: { [ key: string ]: unknown } = {
-            [ GovernanceConnectorUtils.decodeConnectorPropertyName(
-                ServerConfigurationsConstants.USERNAME_RECOVERY_ENABLE) ]: value.toString()
+    const getUpdatedConfigurations = (values: Record<string, any>) => {
+        const data: UsernamePasswordRecoveryFormUpdatableConfigsInterface = {
+            "Recovery.Notification.Username.Email.Enable": values.enableEmailBasedRecovery !== undefined
+                ? values.enableEmailBasedRecovery
+                : initialConnectorValues?.enableEmailBasedRecovery,
+            "Recovery.Notification.Username.Enable":
+                (values.enableEmailBasedRecovery || values.enableSMSBasedRecovery) !== undefined
+                    ? values.enableEmailBasedRecovery || values.enableSMSBasedRecovery
+                    : (initialConnectorValues?.enableEmailBasedRecovery
+                        || initialConnectorValues?.enableSMSBasedRecovery),
+            "Recovery.Notification.Username.SMS.Enable": values.enableSMSBasedRecovery !== undefined
+                ? values.enableSMSBasedRecovery
+                : initialConnectorValues?.enableSMSBasedRecovery
         };
 
         return data;
     };
 
+    if (!initialConnectorValues){
+        return null;
+    }
+
     return (
         <div className={ "connector-form" }>
             <Form
                 id={ FORM_ID }
-                onSubmit={ null }
+                onSubmit={ (values: Record<string, any>) => onSubmit(getUpdatedConfigurations(values)) }
                 uncontrolledForm={ false }
+                initialValues={ initialConnectorValues }
             >
+                <Heading as="h4" className="mb-5">
+                    { t("extensions:manage.serverConfigurations.accountRecovery." +
+                            "passwordRecovery.recoveryOptionHeading") as ReactNode }
```

Comment:
Do we really need to cast to ReactNode

Reply:
Removed from https://github.com/wso2/identity-apps/pull/7004/commits/59d040047dd8b21efd511e781c167e8ebd65b64f